canonical-web-and-design/deployment-configs
# ubuntu-com.internal -> ubuntu.com, add redirects

Instead of listening at ubuntu-com.internal, listen at
ubuntu.com.

The Squid frontends for ubuntu.com will now send the `Host` header
header directly through to k8s. This allows us to do redirects on other
domains, e.g. `ubunut.com -> ubuntu.com`. But for this to work,
this ingress needs to work off the `ubuntu.com` header.

So I'm also adding aliases for the other domains which should be
redirected to ubuntu.com:

- www.ubuntu.com
- [www.]ubunut.com
- [www.]ubuntu.net
- [www.]ubuntulinux.org

QA
--

``` bash
./qa-deploy --production ubuntu.com --tag latest
watch microk8s.kubectl get all,ingress
```

^ Wait for everything to appear to have come up

Now point all relevant domains at localhost:

``` bash
echo "127.0.0.1 ubuntu.com www.ubuntu.com ubuntu.net www.ubuntu.net ubunut.com www.ubunut.com ubuntulinux.org www.ubuntulinux.org" | sudo tee -a /etc/hosts
```

Then try:

``` bash
$ curl -I http://ubuntu.com  # 200
$ curl -I http://www.ubuntu.com  # 301
$ curl -I http://ubuntu.net  # 301
$ curl -I http://www.ubuntu.net  # 301
$ curl -I http://ubunut.com  # 301
$ curl -I http://www.ubunut.com  # 301
$ curl -I http://ubuntulinux.org  # 301
$ curl -I http://www.ubuntulinux.org  # 301
```
